torch/lib/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1164,12 +1164,60 @@ std::shared_ptr<ProcessGroup::Work> ProcessGroupNCCL::alltoall_base(
 }
 #endif
 
+#ifdef ENABLE_NCCL_P2P_SUPPORT
+std::shared_ptr<ProcessGroup::Work> ProcessGroupNCCL::alltoall(
+    std::vector<at::Tensor>& outputTensors,
+    std::vector<at::Tensor>& inputTensors,
+    const AllToAllOptions& /* unused */) {
+  auto device = outputTensors[0].device();
+  for (size_t r = 0; r < outputTensors.size(); r++) {
```

Comment:
Reuse existing checks? 
https://github.com/pytorch/pytorch/blob/37093f4d99c22cb9d89d9eb1cd56b97be92938b4/torch/lib/c10d/ProcessGroupNCCL.cpp#L687

https://github.com/pytorch/pytorch/blob/37093f4d99c22cb9d89d9eb1cd56b97be92938b4/torch/lib/c10d/ProcessGroupNCCL.cpp#L696

Reply:
I think `check_gpu_tensors` checks that all tensors are at different devices, which is different from the use case here where tensors in the list are required to be on the same device.